src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int REQUIRED_AGE = 18;
+    private static final int MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidInputException("No data provided");
+        }
+        loginValidation(user);
+        passwordValidation(user);
+        ageValidation(user);
+        return storageDao.add(user);
+    }
+
+    private void loginValidation(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidInputException("Please, fill all the required fields");
+        }
```

Comment:
this message holds no information at all

isn't it better to say something like "Login is required to fill"

**Make sure to change this in all places**